### DISCUSSION TITLE
Foreach loop with checkboxes is combining each selection

### DISCUSSION THREAD

**Discussion by natewscott** [livewire/livewire#4585] [1. Help]
I am trying to use checkboxes to load a list of size options for a user to add when creating a new item in a database. If I check each box I can see that it is gathering each ID as I would expect it to do. However, when I get the save function I am only getting one array with both the IDs in it instead of two.

Here is the example of the loop in my blade file: 

```
<div>
    <form wire:submit.prevent="save">
        <div class="formrow">
            @foreach($sizes as $index => $size)
            <div class="fromitem"  wire:key="{{$loop->index }}" >

                <label for="size" class="label">
                    {{$size->size_name}} 
                </label>
                <input type="checkbox" 
                name="size"
                wire:model="selected_sizes.{{ $size->id }}" 
                value="{{$size->id}}">
            </div>
            @endforeach
        </div>
        <button type="submit">Add</button>
        @forelse($selected_sizes as $selected_size)
        Sizes: {{$selected_size}}
        @empty
        No size Selected
        @endforelse
    </form>
</div>
```

And here is what I have in my Controller:

```
use App\Models\ProductSize;
use App\Models\ProductInventory;
use Livewire\Component;


class Checkbox extends Component
{
    public $sizes = [];
    public $selected_sizes = [];

    protected $rules = [
        'selected_sizes' => 'nullable',
    ];

    public function render()
    {
        return view('livewire.checkbox');
    }

    public function mount() {
        $this->selected_sizes = $selected_sizes = [];
    }

    public function save() {
        $this->validate();

        foreach ($this->selected_sizes as $index => $size) {
            $data = [
                'size' => $this->size_ids,
             ];
           
            dd($data);
        }
        
    }
}

```
Here is what I see when I check a couple of the boxes:

<img width="1318" alt="Screen Shot 2022-01-28 at 1 44 32 PM" src="https://user-images.githubusercontent.com/6037127/151628203-79fe072b-28ea-4ebd-82d7-d0770cb15b30.png">

Each size ID is listed. Now when I click add here is what I see:
<img width="1481" alt="Screen Shot 2022-01-28 at 1 52 21 PM" src="https://user-images.githubusercontent.com/6037127/151628361-bc1d6d03-b42b-46dc-9287-e1fa1c68a73c.png">

I am new to using Livewire, but I would expect that this would give me two arrays to be inserted into the database. What am I missing here? 

Thanks for any help in advance.

**Comment by dennisprudlo** [comment]
Have you tried `wire:model="selected_sizes"` instead of `wire:model="selected_sizes.{{ $size->id }}"`. Livewire should be smart enough to know that the list of checkboxes have to be an array.

**Reply by natewscott** [reply]
@dennisprudlo, yes I have tried that as well. It gives me the same results.

**Comment by silnarm** [comment]
Livewire can't possibly know what you intend to do with the data you have bound, it just puts it where you tell it.
In this case you are telling it to put each value in the same array (selected_sizes), each under it's own key ($size->id):
`wire:model="selected_sizes.{{ $size->id }}"`
So there is nothing unexpected here.
The save method you posted is a bit bizarre, looping over an array but ignoring the key and value, and with a dd() in the loop (dd means 'dump and die', so you are never getting to the second iteration of the loop, if there would be one).
It is also referencing an undefined (in the code you pasted) property.
If you want the Ids in separate arrays, keyed by some specific (column) name ('ready to insert') you can do something like this:
```
    public function save() {
        $this->validate();
        $data = [];
        foreach ($this->selected_sizes as $size) {
            $data[] = ['size_id' => $size];
        }
        dd($data);
    }
```
If you want the data to be in that form in the first place, you will have to tell Livewire to put it there,
`wire:model="selected_sizes.{{ $size->id }}.size_id"`
Which would of course require the selected_sizes array to be structured accordingly. Doing it in the save method as above would be easier.
